MyApplication/app/src/main/java/com/courses/bogdan/myapplication/activities/MainActivity.java

```diff
@@ -15,86 +17,164 @@
-                green = randomColor.nextInt(256);
-                blue = randomColor.nextInt(256);
-                layout.setBackgroundColor(Color.argb(255,red,green,blue));
+    @Override
+    protected void onPause() {
+        super.onPause();
+        unregisterReceiver(myReceiver);
+    }
 
+    private void registerBroadcastReceiver() {
+        IntentFilter intentFilter = new IntentFilter();
+        intentFilter.addAction(IntentActionConstants.ACTIVITY_INFO_INTENT);
+        registerReceiver(myReceiver, intentFilter);
+    }
+
+    private void setupViews() {
+        mainlayout = (RelativeLayout) findViewById(R.id.layout);
+        toastButton = (Button) findViewById(R.id.toastButton);
+        getButton = (Button) findViewById(R.id.getTextButton);
+        colorButton = (Button) findViewById(R.id.colorButton);
+        fragmentButton = (Button) findViewById(R.id.fragmentGo);
+        myView = (TextView) findViewById(R.id.myText);
+        textEditor = (EditText) findViewById(R.id.textEditor);
+
+        toastButton.setOnClickListener(toastButtonListener);
+        getButton.setOnClickListener(getButtonListener);
+        colorButton.setOnClickListener(colorButtonListener);
+        fragmentButton.setOnClickListener(fragmentButtonListener);
+    }
 
+    private void setupFragment() {
+        MainFragment fragment = MainFragment.newInstance();
+        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+        transaction.add(R.id.main_frame_layout, fragment).commit();
+    }
+
+    private void setupListeners() {
+        colorButtonListener = new ColorButtonListener();
+        fragmentButtonListener = new FragmentButtonListener();
+        getButtonListener = new GetButtonListener();
+        toastButtonListener = new ToastButtonListener();
+    }
+
+    private void setTextToTextView(final String text) {
+        this.runOnUiThread(new Runnable() {
+            public void run() {
+                myView.setText(text);
             }
         });
-
     }
 
     @Override
-    protected void onResume() {
-        super.onResume();
-        Random randomColor = new Random();
-        red= randomColor.nextInt(256);
-        green = randomColor.nextInt(256);
-        blue = randomColor.nextInt(256);
-        layout.setBackgroundColor(Color.argb(255,red,green,blue));
-        getButton.setEnabled(false);
+    public void setMainLayoutBackgroundColor(int color) {
+        mainlayout.setBackgroundColor(color);
     }
 
-    @Override
-    protected void onStart() {
-        super.onStart();
-        Random randomColor = new Random();
-        red= randomColor.nextInt(256);
-        green = randomColor.nextInt(256);
-        blue = randomColor.nextInt(256);
-        layout.setBackgroundColor(Color.argb(255,red,green,blue));
+    class MyReceiver extends BroadcastReceiver {
+
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            Log.e(TAG, "dvniv");
```

Comment:
E bine ceea ce ai facut, insa, pe viitor te incurajez ca sa treci mesaje relevante in log-uri. De exemplu, aici poate vrei sa verifici in log-uri daca s-a executat metoda de onReceive. Ca urmare, ca si mesaj poti seta ceva de genul: "onReceive method called", deoarece ajuta pe altii sa inteleaga log-urile, altfel raman confuzi cand vad mesajele "dvniv" printre log-uri deoarece pentru ei aceste mesaje nu spun nimic. :)